roles/ansible-test-splitter/files/k8s_list_changed_targets.py

```diff
@@ -0,0 +1,291 @@
+            return True
+        return False
+
+    def is_ignored(self):
+        """Show the target be ignored by default?"""
+        ignore = set(["unsupported", "disabled", "unstable", "hidden"])
+        return not ignore.isdisjoint(set(self.lines))
+
+
+class Collection:
+    def __init__(self, path):
+        self.collection_path = path
+        self._my_test_plan = []
+        self.collection_name = read_collection_name(path)
+
+        self.timers = {}
+        timer_file = PosixPath(path / "tests/integration/timer.json")
+        if timer_file.exists():
+            with timer_file.open() as fd:
+                self.timers = json.loads(fd.read())
+
+    def _targets(self):
+        for a in self.collection_path.glob("tests/integration/targets/*/aliases"):
+            yield Target(a)
+
+    def add_target_to_plan(self, target_name):
+        for t in self._targets():
+            if t.is_disabled():
+                continue
+            if t.is_alias_of(target_name):
+                self._my_test_plan.append(t)
+
+    def cover_all(self):
+        """Cover all the targets available."""
+        for t in self._targets():
+            if t.is_ignored():
+                continue
+            self.add_target_to_plan(t.name)
+
+    def add_target_for_module(self, k8smod):
+        """Track the targets to run follow up to a module changed."""
+        self.add_target_to_plan(f"module/{k8smod}")
+
+    def split_target_per_jobs(self, nb_jobs, releases):
+        if len(self._my_test_plan) == 0:
+            self.cover_all()
+
+        targets = [(t.name, self.timers.get(t.name, 0)) for t in self._my_test_plan]
+        sorted_targets = sorted(targets, key=lambda x: x[1], reverse=True)
+
+        slots = [{"total": 0, "targets": []} for _ in range(nb_jobs)]
+
+        def _get_index_lowest():
+            lowest = 0
+            idx = 1
+            while idx < len(slots):
+                if slots[idx]["total"] < slots[lowest]["total"]:
+                    lowest = idx
+                idx += 1
+            return lowest
+
+        for t in sorted_targets:
+            insert_at = _get_index_lowest()
+            slots[insert_at]["total"] += t[1]
+            slots[insert_at]["targets"].append(t[0])
+
+        # build result
+        result = {
+            "data": {
+                "zuul": {"child_jobs": []},
+                "child": {"targets_to_test": {}},
+            }
+        }
+
+        job_id = 1
+        for slot_id in slots:
+            for rel in releases:
+                job_name = f"integration-{self.collection_name}-{rel}-{job_id}"
+                result["data"]["zuul"]["child_jobs"].append(job_name)
+                result["data"]["child"]["targets_to_test"][job_name] = " ".join(
+                    slot_id["targets"]
+                )
+            job_id += 1
+        return result
+
+
+if __name__ == "__main__":
+    args = parse_args(sys.argv[1:])
+
+    collection = Collection(args.collection_to_test)
```

Comment:
I believe it can be handy in the future to test community.okd and kubernetes.core at the same time. I would keep the ability to have a list of collections.